src/release/rollups.md

```diff
@@ -23,67 +23,99 @@ queue has been merged.
 ## Making a Rollup
 
 1. Using the interface on [Homu queue], select pull requests and then
-   use "rollup" button to make a rollup pull request. (The text about
-   fairness can be ignored.)
+   use "rollup" button to make a rollup pull request.
+
    **Important note**:  consider for addition PRs marked as
    `rollup=always`, `rollup=maybe` and `rollup=iffy`, based on the
    review policies of [the Rollups section].  Be extra careful when
    deciding what to include, in particular on `rollup=maybe` and
    `rollup=iffy` PRs. We should try as much as possible to avoid risking
-   and hit regressions (bugs or perf).  Also consider that contributors
+   to hit regressions (bugs or perf).  Also consider that contributors
    often forget to tag things with rollup=never, when they should have
    done so, so when PRs are not explicitly tagged with rollup, be extra
-   careful.
+   careful. Also carefully consider the area of the compiler the PRs touch.
+   Two diagnostic PRs may actually conflict with each other, as they both 
+   change compiler output, which causes failures in each other's tests, 
+   when both of them are merged together in a rollup without causing git merge-conflicts.
+   In this case the older PR should be given the privilege to merge first 
+   and the newer PR should then be rebased as needed.
 
 2. Run the following command in the pull request thread:
 
     ```
     @bors r+ rollup=never p=5
     ````
-
+   where 5 is the number of PRs contained in your rollup.
```

Comment:
I have been thinking that what the existing documentation says (rollups always p=5) might be better than p=pr_count that is usually done. There are a lot of cases where 7 PRs get rolled up with p=7 and sit in the queue for a number of hours, only to be leapfrogged by a p=8 rollup that contains much newer PRs. Seems to have a tendency to break FIFO.

Reply:
Yes, I noticed a certain someone was doing this a lot. That's why I added the "Rollups should not overlap" rule.

Reply:
> Yes, I noticed a certain someone was doing this a lot. That's why I added the "Rollups should not overlap" rule.

Again, there's no need to throw shade when we can just ping the contributors who are assigning rollup p>5 (I've done this a couple of times myself) and just point them to p=5 being the good default and having that respect queue order. It helps to elaborate on the choice of p=5 and not arbitrary p=N where N is the number of PRs rolled up.

Reply:
(Thanks for the edits)

Just wanted to say, we should make these rollup advice as accessible and useful to onboard other contributors w/ r+ perms who may want to help to do rollups too.

Reply:
Maybe we should have something like
tool/subtree updates `p=5` and rollups `p=5` as default, so that we still have some headroom to `4` or `3` things if needed

Reply:
Yeah, I feel like the tool/subtree updates should receive same priority as rollups and all pinned at `p=5` exactly, good point.

Reply:
mmh imo it makes sense to prioritise subtree/module updates over rollups since doing a tool sync is often more complicated than rebasing PRs

Reply:
Just, they should receive a consistent `p` between themselves, but otherwise it feels whatever to me.